src/invidious/search.cr

```diff
@@ -231,20 +231,27 @@ end
 alias SearchItem = SearchVideo | SearchChannel | SearchPlaylist
 
 def channel_search(query, page, channel)
-  response = YT_POOL.client &.get("/channel/#{channel}?hl=en&gl=US")
-  response = YT_POOL.client &.get("/user/#{channel}?hl=en&gl=US") if response.headers["location"]?
-  response = YT_POOL.client &.get("/c/#{channel}?hl=en&gl=US") if response.headers["location"]?
+  response = YT_POOL.client &.get("/channel/#{channel}")
 
-  ucid = response.body.match(/\\"channelId\\":\\"(?<ucid>[^\\]+)\\"/).try &.["ucid"]?
+  if response.status_code == 404
+    response = YT_POOL.client &.get("/user/#{channel}")
+    response = YT_POOL.client &.get("/c/#{channel}") if response.status_code == 404
+    ucid = response.body.match(/HeaderRenderer":\{"channelId":"(?<ucid>[^\\"]+)"/).try &.["ucid"]?
```

Comment:
Is this in the initial data?

Reply:
Yes, let me guess
you would rather I process the entire `ytInitialData` object as JSON?